packages/stark-ui/src/modules/route-search/components/route-search.component.ts

```diff
@@ -0,0 +1,287 @@
+
+/**
+ *  @ignore
+ */
+const _sortBy: Function = require("lodash/sortBy");
+/**
+ * Name of the component
+ *
+ */
+const componentName: string = "stark-route-search";
+
+/**
+ * Direction in which the Route Search field can be displayed
+ */
+export type StarkRouteSearchDirection = "left" | "right";
+
+/**
+ * Component to search a particular route with autocompletion and to redirect to the route chosen by the user
+ */
+@Component({
+	selector: "stark-route-search",
+	templateUrl: "./route-search.component.html",
+	encapsulation: ViewEncapsulation.None,
+	host: {
+		class: componentName
+	}
+})
+export class StarkRouteSearchComponent implements OnInit {
+	/**
+	 * The source FormControl object of the search field
+	 */
+	public searchField: FormControl;
+
+	/**
+	 * The list of {@link RouteEntry}'s objects filtered by the auto-complete field
+	 */
+	public filteredRouteEntries: Observable<RouteEntry[]>;
+
+	/**
+	 * The list of routes to be displayed
+	 */
+	public routesToDisplay: RouteEntry[] = [];
+
+	/**
+	 * The {@link StarkMenuConfig} where the routes should be taken from.
+	 * Normally the same {@link StarkMenuConfig} is used for the {@link StarkAppMenuComponent}.
+	 */
+	@Input()
+	public menuConfig?: StarkMenuConfig;
+
+	/**
+	 * The direction in which the Route Search field will be displayed
+	 */
+	@Input()
+	public direction: StarkRouteSearchDirection = "left";
+
+	/**
+	 * Whether the input field must be hidden or not
+	 */
+	public hide: boolean;
+
+	/**
+	 * Class constructor
+	 * @param starkRoutingService - The routing service of the application
+	 * @param logger - The logger of the application
+	 * @param translateService -  The translation service of the application
+	 * @param elementRef- Reference to the DOM element where this directive is applied to.
+	 */
+	public constructor(
+		@Inject(STARK_ROUTING_SERVICE) public starkRoutingService: StarkRoutingService,
+		@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService,
+		@Inject(TranslateService) public translateService: TranslateService,
+		public elementRef: ElementRef
+	) {
+		this.searchField = new FormControl();
+	}
+
+	/**
+	 * Filters the options according to what has been entered by the user in the input field
+	 */
+	public filterRouteEntries(value: string): RouteEntry[] {
+		const filterValue: string = value.toLowerCase();
+		return this.routesToDisplay.filter(
+			(state: RouteEntry) =>
+				state.label
+					.toString()
+					.toLowerCase()
+					.indexOf(filterValue) === 0
+		);
+	}
+
+	public ngOnInit(): void {
+		this.searchField.setValue("");
+		this.filteredRouteEntries = this.searchField.valueChanges.pipe(
+			startWith(""),
+			map((option: string) => {
+				return option ? this.filterRouteEntries(option) : [...this.routesToDisplay];
+			})
+		);
+
+		this.hide = true;
+
+		this.logger.debug(componentName + ": component initialized");
+
+		if (this.routesToDisplay !== undefined) {
```

Comment:
According to what I see here, I guess you already thought to set `routesToDisplay` as an Input binding 😊

Reply:
As discussed, we'll leave that as a "nice to have" for later ;)